tests/ipfwd/test_nhop_group.py

```diff
@@ -371,3 +417,108 @@ def test_nhop(request, duthost, tbinfo):
                 crm_after["available"], crm_after["used"]
             )
         )
+
+def test_nhop_group_member_order_capability(request, duthost, tbinfo, ptfadapter, gather_facts, enum_rand_one_frontend_asic_index):
+    """
+    Test SONiC and SAI Vendor capability are same for ordered ecmp feature
+    and SAI vendor is honoring the Ordered nature of nexthop group member
+    """
+    asic = duthost.asic_instance(enum_rand_one_frontend_asic_index)
+
+    result = asic.run_redis_cmd(
+        argv = ["redis-cli", "-n", 6, "HGETALL", "SWITCH_CAPABILITY|switch"]
+    )
+    it = iter(result)
+    switch_capability = dict(zip(it, it))
+
+    result = asic.run_redis_cmd(
+            argv = ["redis-cli", "-n", 0, "HGETALL", "SWITCH_TABLE:switch"]
+    )
+
+    it = iter(result)
+    switch_table = dict(zip(it, it))
+
+    order_ecmp_capability = switch_capability.get("ORDERED_ECMP_CAPABLE")
+    order_ecmp_configured = switch_table.get("ordered_ecmp")
+    pytest_assert(order_ecmp_capability == order_ecmp_configured, "Order Ecmp Feature configured and capability not same")
+
+    if order_ecmp_configured == "false":
+        pytest.skip("Order ECMP is not configured so skipping the test-case")
+
+    # Check Gather facts IP Interface is active one
+    ip_ifaces = asic.get_active_ip_interfaces(tbinfo).keys()
+    pytest_assert(len(ip_ifaces), "No IP interfaces found")
+    pytest_assert(gather_facts['src_router_intf_name'] in ip_ifaces, "Selected IP interfaces is not active")
+
+    # Generate ARP entries
+    arp_count = 8
+    arplist = Arp(duthost, asic, arp_count, gather_facts['src_router_intf_name'])
+    neighbor_mac = [neighbor[1].lower() for neighbor in arplist.ip_mac_list]
+    original_ip_mac_list = arplist.ip_mac_list[:]
+    ip_route = "192.168.100.50"
+    ip_prefix = ip_route + "/31"
+    ip_ttl = 121
+
+    # create nexthop group
+    nhop = IPRoutes(duthost, asic)
+
+    recvd_pkt_result = defaultdict(int)
+
+    rtr_mac= asic.get_router_mac()
+    
+    pkt, exp_pkt = build_pkt(rtr_mac, ip_route, ip_ttl)
+
+    for x in range(2):
+        try:
+            # create neighbor entry randomly
+            random.shuffle(arplist.ip_mac_list)
```

Comment:
randomness may cause unpredictable results across runs. Can we create this manually in a shuffled order but always consistent across runs.

Reply:
@prsunny done. Setting up seed to initialize random generator so we get same order in every test run